yolo/tools/data_loader.py

```diff
@@ -100,7 +109,7 @@ def filter_data(self, dataset_path: Path, phase_name: str) -> list:
 
             labels = self.load_valid_labels(image_id, image_seg_annotations)
 
-            img_path = images_path / image_name
+            img_path = Path("images") / phase_name / image_name
             data.append((img_path, labels))
```

Comment:
Instead of `image_path` can we consider using `image_id` as the key?
That is instead do this:
`data.append((image_id, labels))` where `image_id = Path(image_name).stem`
This should meet the requirement of this MR that we can move around dataset randomly.
This also makes the code a bit more predictable. I think it is more intuitive that a cache has keys that are just file names and not the information about train / test split. 
The current check for cache is just checking whether the train/test split are same as `phase_name`: `if data[0][0].parent == Path("images")/phase_name:`. This will no longer be needed. Please also see my comment about this line inside the `load_data` function.

Reply:
It's a good idea that solved the input info and the question of validation.